imap_processing/swapi/l1/swapi_l1.py

```diff
@@ -1 +1,520 @@
+    |    PCEM_CNT2 --> 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11
+    |    PCEM_CNT3 --> 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11
+    |    PCEM_CNT4 --> 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11
+    |    PCEM_CNT5 --> 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11
+
+    This example show for PCEM_CNT but same logic applies
+    to SCEM_CNT, COIN_CNT, PCEM_RNG, SCEM_RNG, and COIN_RNG.
+
+    In the final L1A product of (total_number_of_sweep x 72) array where
+    we store final PCEM, SCEM, COIN counts or compression indicator
+    such as PCEM_RNG, SCEM_RNG, COIN_RNG,
+    we want data in this order. Transpose of above layout
+
+    |   0, 0, 0, 0, 0, 0,
+    |   1, 1, 1, 1, 1, 1,
+    |   2, 2, 2, 2, 2, 2,
+    |   3, 3, 3, 3, 3, 3,
+    |   ....,
+    |   11, 11, 11, 11, 11, 11.
+
+    Reordering in this order is reordering all data of
+    sequence 0 first, then sequence 1, then sequence 2,
+    and so on until sequence 11.
+
+    Parameters
+    ----------
+    full_sweep_sci : xarray.Dataset
+        Full dataset
+    cem_prefix : str
+        Indicate which CEM or its flag we are processing. Options are:
+
+        |    PCEM_CNT
+        |    SCEM_CNT
+        |    COIN_CNT
+        |    PCEM_RNG_ST
+        |    SCEM_RNG_ST
+        |    COIN_RNG_ST
+    total_full_sweeps: int
+        Total number of sweeps
+    """
+    all_cem_data = np.zeros((total_full_sweeps, 72))
+
+    # First, concat all CEM data
+    current_cem_counts = np.concatenate(
+        (
+            full_sweep_sci[f"{cem_prefix}0"],
+            full_sweep_sci[f"{cem_prefix}1"],
+            full_sweep_sci[f"{cem_prefix}2"],
+            full_sweep_sci[f"{cem_prefix}3"],
+            full_sweep_sci[f"{cem_prefix}4"],
+            full_sweep_sci[f"{cem_prefix}5"],
+        ),
+        axis=0,
+    )
+
+    # Next:
+    # Reshape data by CEM, number of sweeps and sequence counts.
+    # Therefore, the data shape is 6 x total_full_sweeps x 12
+    # Output looks like this:
+    # [
+    # [[ 0  1  2  3  4  5  6  7  8  9 10 11]
+    # [ 1  2  3  4  5  6  7  8  9  10  11  12]
+    # [ 2  3  4  5  6  7  8  9  10  11  12  13]]
+
+    # [[ 0  1  2  3  4  5  6  7  8  9 10 11]
+    # [ 1  2  3  4  5  6  7  8  9  10  11  12]
+    # [ 2  3  4  5  6  7  8  9  10  11  12  13]]
+
+    # [[ 0  1  2  3  4  5  6  7  8  9 10 11]
+    # [ 1  2  3  4  5  6  7  8  9  10  11  12]
+    # [ 2  3  4  5  6  7  8  9  10  11  12  13]]
+
+    # [[ 0  1  2  3  4  5  6  7  8  9 10 11]
+    # [ 1  2  3  4  5  6  7  8  9  10  11  12]
+    # [ 2  3  4  5  6  7  8  9  10  11  12  13]]
+
+    # [[ 0  1  2  3  4  5  6  7  8  9 10 11]
+    # [ 1  2  3  4  5  6  7  8  9  10  11  12]
+    # [ 2  3  4  5  6  7  8  9  10  11  12  13]]
+
+    # [[ 0  1  2  3  4  5  6  7  8  9 10 11]
+    # [ 1  2  3  4  5  6  7  8  9  10  11  12]
+    # [ 2  3  4  5  6  7  8  9  10  11  12  13]]]
+    # In other word, we grouped each cem's
+    # data by full sweep.
+    current_cem_counts = current_cem_counts.reshape(6, total_full_sweeps, 12)
```

Comment:
I think you can put a -1 in here to allow numpy to do this reshaping for you without needing to pass in the integer of the total_full_sweeps.
```suggestion
    current_cem_counts = current_cem_counts.reshape(6, -1, 12)
```